File: src/main/java/examples/GreetingActor_1.java
package examples;

import akka.actor.AbstractActor;
import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;

public class GreetingActor_1 extends AbstractActor {

    @Override
    public Receive createReceive() {
        return receiveBuilder()
            .match(String.class, message -> {
                System.out.println("Hello " + message);
            }).build();
    }

    public static void main(String[] args) throws InterruptedException {
        ActorSystem system = ActorSystem.create("MySystem");

        ActorRef greetingACtor = system.actorOf(Props.create(GreetingActor_1.class), "greeter");
        greetingACtor.tell("Hulk Hogan", ActorRef.noSender());

        // There are better ways to ensure message are received before termination
        Thread.sleep(100);
        system.terminate();
    }
}

File: src/main/kotlin/examples/GreetingActor_1.kt
package examples

import akka.actor.ActorRef
import akka.actor.ActorSystem
import akka.actor.Props
import akka.actor.UntypedActor

class GreetingActor_1 : UntypedActor() {

    override fun onReceive(msg: Any?) = when (msg) {
        is String -> {
            System.out.println("Hello " + msg)
        }
        else -> {
            unhandled(msg)
        }
    }
}

fun main(args: Array<String>) {
    val system = ActorSystem.create("MySystem")

    val greetingACtor = system.actorOf(Props.create(GreetingActor_1::class.java), "greeter")
    greetingACtor.tell("Hulk Hogan", ActorRef.noSender())

    // There are better ways to ensure message are received before termination
    Thread.sleep(100)
    system.terminate()
}
